src/Handler/CommandNameExtractor/NamedCommandExtractor.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace League\Tactician\Handler\CommandNameExtractor;
+
+/**
+ * Extract the name from a NamedCommand
+ */
+class NamedCommandExtractor implements CommandNameExtractor
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getNameForCommand($command)
+    {
+        if ($command instanceof NamedCommand) {
+            return $command->getName();
+        }
+
+        // fallback to class name
+        return get_class($command);
```

Comment:
Exception name variations (based on current exceptions):

- CanNotDetermineCommandName
- CanNotDetermineCommandNameException
- CommandNameCouldNotBeDetermined

(Just to make sure we remain consistent. Also, I am pretty against the `Exception` suffix, especially if we place exceptions under a separate namespace.